packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -409,18 +409,26 @@ export function updateTableForConnection(config: HasManyDirectiveConfiguration,
   // At the L2 level, the CDK does not handle the way Amplify sets GSI read and write capacity
   // very well. At the L1 level, the CDK does not create the correct IAM policy for accessing the
   // GSI. To get around these issues, keep the L1 and L2 GSI list in sync.
-  const cfnTable = table.table;
   const gsi = gsis.find((gsi: any) => gsi.indexName === indexName);
 
-  cfnTable.globalSecondaryIndexes = appendIndex(cfnTable.globalSecondaryIndexes, {
+  const newIndex = {
     indexName,
     keySchema: gsi.keySchema,
     projection: { projectionType: 'ALL' },
     provisionedThroughput: cdk.Fn.conditionIf(ResourceConstants.CONDITIONS.ShouldUsePayPerRequestBilling, cdk.Fn.ref('AWS::NoValue'), {
       ReadCapacityUnits: cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBModelTableReadIOPS),
       WriteCapacityUnits: cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBModelTableWriteIOPS),
     }),
-  });
+  };
+
+  if (!ctx.transformParameters.useAmplifyManagedTableResources) {
+    const cfnTable = table.table;
+    cfnTable.globalSecondaryIndexes = appendIndex(cfnTable.globalSecondaryIndexes, newIndex);
+  } else {
+    const cfnTable = table.node.defaultChild.node.defaultChild as cdk.CfnCustomResource;
+    const idx = table.globalSecondaryIndexes.length - 1;
```

Comment:
Why does only the last GSI need to be replaced?

Reply:
The replacement here is mainly for changing the `provisionThroughput` attribute of index to the dynamic one, which will use the cloudformation condition(for the billing mode) to determine the value instead of the hard-code default value.